src/Gaufrette/Adapter/Ftp.php

```diff
@@ -54,93 +55,122 @@ public function __construct($directory, $host, $options = array())
     }
 
     /**
-     * {@inheritDoc}
+     * @param string $key
+     * @return bool|string
      */
     public function read($key)
     {
-        $this->ensureDirectoryExists($this->directory, $this->create);
+        //change to the directory that contain the file
+        $this->moveToTargetDirectory($key);
+        $file = basename($key);
 
         $temp = fopen('php://temp', 'r+');
 
-        if (!ftp_fget($this->getConnection(), $temp, $this->computePath($key), $this->mode)) {
```

Comment:
Indeed, this code is from early this year. I wanted to know if you would change the feature completely before updating it. I will make the changes on another branch.
Thanks for the review.